src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -10,19 +10,32 @@ namespace Microsoft.Docs.Build
 {
     internal class PublishModelBuilder
     {
+        private readonly ErrorLog _errorLog;
+
         private readonly ConcurrentDictionary<string, ConcurrentBag<Document>> _outputPathConflicts = new ConcurrentDictionary<string, ConcurrentBag<Document>>(PathUtility.PathComparer);
         private readonly ConcurrentDictionary<string, ConcurrentDictionary<Document, List<string>>> _filesBySiteUrl = new ConcurrentDictionary<string, ConcurrentDictionary<Document, List<string>>>(PathUtility.PathComparer);
         private readonly ConcurrentDictionary<string, Document> _filesByOutputPath = new ConcurrentDictionary<string, Document>(PathUtility.PathComparer);
         private readonly ConcurrentDictionary<Document, PublishItem> _publishItems = new ConcurrentDictionary<Document, PublishItem>();
         private readonly ListBuilder<Document> _filesWithErrors = new ListBuilder<Document>();
 
+        public PublishModelBuilder(ErrorLog errorLog)
+        {
+            _errorLog = errorLog;
+        }
+
         public void MarkError(Document file)
         {
             _filesWithErrors.Add(file);
         }
 
-        public bool TryAdd(Document file, PublishItem item)
+        public bool TryAdd(Document file, IEnumerable<Error> errors, PublishItem item)
```

Comment:
>IEnumerable<Error> errors [](start = 42, length = 25)

could you split this change into another PR then in this PR we can focus on the change of `BuildChildren`?

Reply:
ok